### ISSUE TITLE
Wrong dimension in torch.sum for Attention Layer

### ISSUE THREAD

**Issue by sidr97** [alecwangcq/show-attend-and-tell#6]
In the attention layer, the sum while calculating context should be along the 2nd axis and not the 1st, because output has to be weights for each area to focus on, hence (batch_size x 196), but in the code, output is (batch_size x 512)
